From 7767900679134b553b081c072c6b83bfe10cc0ef Mon Sep 17 00:00:00 2001
From: Megha Aggarwal <megha.aggarwal@intel.com>
Date: Tue, 25 Aug 2015 00:37:09 +0530
Subject: [PATCH] FOR_UPSTREAM [VPG]: drm/i915: Command mode Support with
 IMINAN- 39625

To optimize the blank out time during boot, the modeset was by-passed
in driver if already done in the BIOS. In case of command-mode,
the BIOS bu default configuring for video mode, hence modeset
is required in the driver to support command mode.
The check has been added to by-pass modeset only in video mode.

Change-Id: I2a5da9b7cd30dcaf2fa555be48dd16d84fae2377
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7482
Signed-off-by: Megha Aggarwal <megha.aggarwal@intel.com>
Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index ba8a8d8..7ff0d61 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5538,7 +5538,9 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 	for_each_encoder_on_crtc(dev, crtc, encoder) {
 		if (encoder->pre_pll_enable) {
 			if (encoder->type == INTEL_OUTPUT_DSI) {
-				if (dev_priv->quick_modeset)
+				intel_dsi = enc_to_intel_dsi(&encoder->base);
+				if (dev_priv->quick_modeset &&
+							is_vid_mode(intel_dsi))
 					continue;
 			}
 			encoder->pre_pll_enable(encoder);
@@ -5554,7 +5556,9 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 
 	for_each_encoder_on_crtc(dev, crtc, encoder) {
 		if (encoder->pre_enable) {
-			if (encoder->type == INTEL_OUTPUT_DSI) {
+			intel_dsi = enc_to_intel_dsi(&encoder->base);
+			if ((encoder->type == INTEL_OUTPUT_DSI) &&
+						is_vid_mode(intel_dsi)) {
 				if (dev_priv->quick_modeset)
 					continue;
 			}
@@ -5598,7 +5602,8 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 
 	for_each_encoder_on_crtc(dev, crtc, encoder) {
 		if (encoder->type == INTEL_OUTPUT_DSI) {
-			if (dev_priv->quick_modeset)
+			intel_dsi = enc_to_intel_dsi(&encoder->base);
+			if (dev_priv->quick_modeset && is_vid_mode(intel_dsi))
 				continue;
 			encoder->enable(encoder);
 		}
@@ -5838,7 +5843,8 @@ static void i9xx_crtc_disable(struct drm_crtc *crtc)
 
 	for_each_encoder_on_crtc(dev, crtc, encoder) {
 		if (encoder->type == INTEL_OUTPUT_DSI) {
-			if (dev_priv->quick_modeset)
+			intel_dsi = enc_to_intel_dsi(&encoder->base);
+			if (dev_priv->quick_modeset && is_vid_mode(intel_dsi))
 				continue;
 		}
 		encoder->disable(encoder);
@@ -5863,7 +5869,9 @@ static void i9xx_crtc_disable(struct drm_crtc *crtc)
 	for_each_encoder_on_crtc(dev, crtc, encoder) {
 		if (encoder->post_disable) {
 			if (encoder->type == INTEL_OUTPUT_DSI) {
-				if (dev_priv->quick_modeset)
+				intel_dsi = enc_to_intel_dsi(&encoder->base);
+				if (dev_priv->quick_modeset &&
+							is_vid_mode(intel_dsi))
 					continue;
 			}
 			encoder->post_disable(encoder);
-- 
1.9.1

